src/app/public/modules/toggle/toggle.component.ts

```diff
@@ -0,0 +1,196 @@
+
+import {
+  Component,
+  EventEmitter,
+  forwardRef,
+  Input,
+  Output
+} from '@angular/core';
+
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+  NG_VALIDATORS
+} from '@angular/forms';
+
+/**
+ * Monotonically increasing integer used to auto-generate unique ids for toggle components.
+ */
+let nextId = 0;
+
+/**
+ * Provider Expression that allows sky-toggle to register as a ControlValueAccessor.
+ * This allows it to support [(ngModel)].
+ */
+// tslint:disable:no-forward-ref no-use-before-declare
+export const SKY_TOGGLE_CONTROL_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyToggleComponent),
+  multi: true
+};
+
+const SKY_TOGGLE_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyToggleComponent),
+  multi: true
+};
+
+// A simple change event emitted by the SkyToggle component.
+export class SkyToggleChange {
```

Comment:
This normally would go into a `types` folder. See https://github.com/blackbaud/skyux-forms/tree/master/src/app/public/modules/radio/types